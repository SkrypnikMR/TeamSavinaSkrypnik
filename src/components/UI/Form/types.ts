import React from 'react';

export interface IStForm {
    color?: string;
    transition?: string;
}
export interface IProps {
  children: React.ReactNode[] | React.ReactNode
}
