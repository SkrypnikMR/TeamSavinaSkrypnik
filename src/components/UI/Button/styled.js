import styled from 'styled-components';
import {
  bgColorDefault,
  colorDefault,
  focusColorDefault,
  disabledColor,
  bgColorDisabledDefault,
  disabledBorder,
} from '../baseLayout';

export const StButton = styled.button`
  ${({ margin = '0 0 3px 0' }) => margin && `margin: ${margin}`};
  font-family: 'Play', sans-serif;
  padding: ${({ padding = '5px' }) => padding};
  width: ${({ width = '170px' }) => width};
  height: ${({ height = '35px' }) => height};
  ${({ cursor = 'pointer' }) => cursor && `cursor: ${cursor}`};
  color: ${({ color = colorDefault }) => color};
  background-color: ${({ bgColor = bgColorDefault }) => bgColor};
  border-radius: ${({ borderRadius = '7px' }) => borderRadius}; 
  border: ${({ border = 'none' }) => border};
  outline: ${({ outline = 'none' }) => outline};
  ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
  font-size: ${({ fontSize = '16px' }) => fontSize};
  &:hover {
    color: ${({ focusColor = focusColorDefault }) => focusColor}; 
    border: ${({ border = 'none' }) => border};
    }
  &:focus {
    color: ${({ focusColor = focusColorDefault }) => focusColor};
    }
  &:disabled{ 
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = bgColorDisabledDefault }) => bgColorDisabled};
    color: ${({ focusColor = disabledColor }) => focusColor};
    }
`;
