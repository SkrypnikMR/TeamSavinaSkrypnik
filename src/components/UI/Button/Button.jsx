import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { StButton } from './styled';

const Button = ({
    id,
    name,
    onClick,
    title,
    type = 'button',
    value,
    borderRadius,
    isDisabled,
    color,
    fontSize,
    fontWeight,
    bgColorDisabled,
    bgColor,
    bgColorHover,
    height,
    width,
    padding,
    margin,
    content,
    transition,
    focusColor,
}) => {
    const { t } = useTranslation();
    return (
        <StButton
            id={id}
            name={name}
            onClick={onClick}
            type={type}
            value={value}
            title={title}
            bgColorDisabled={bgColorDisabled}
            borderRadius={borderRadius}
            isDisabled={isDisabled}
            color={color}
            fontSize={fontSize}
            fontWeight={fontWeight}
            bgColor={bgColor}
            bgColorHover={bgColorHover}
            height={height}
            width={width}
            padding={padding}
            margin={margin}
            transition={transition}
            focusColor={focusColor}
        >
            {t(content)}
        </StButton>
    );
};

Button.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    title: PropTypes.string,
    borderRadius: PropTypes.string,
    isDisabled: PropTypes.string,
    color: PropTypes.string,
    fontSize: PropTypes.string,
    fontWeight: PropTypes.string,
    bgColor: PropTypes.string,
    bgColorHover: PropTypes.string,
    height: PropTypes.string,
    width: PropTypes.string,
    padding: PropTypes.string,
    margin: PropTypes.string,
    content: PropTypes.string,
    transition: PropTypes.string,
    focusColor: PropTypes.string,
    bgColorDisabled: PropTypes.string,
};

export default Button;
