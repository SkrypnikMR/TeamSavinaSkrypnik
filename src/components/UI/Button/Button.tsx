import React from 'react';
import { useTranslation } from 'react-i18next';
import { StButton } from './styled';
import { useTheme } from '../../Hook/useTheme';
import { IButton } from './types';

const Button = ({
    id,
    name,
    onClick,
    title,
    type = 'button',
    value,
    borderRadius,
    isDisabled,
    color,
    fontSize,
    fontWeight,
    bgColorDisabled,
    bgColor,
    bgColorHover,
    height,
    width,
    padding,
    margin,
    content,
    transition,
    focusColor,
    position,
    right,
} : IButton) => {
    const { t } = useTranslation();
    const { colors, theme } = useTheme();

    return (
        <StButton
            id={id}
            name={name}
            onClick={onClick}
            type={type}
            value={value}
            title={title}
            bgColorDisabled={bgColorDisabled}
            borderRadius={borderRadius}
            isDisabled={isDisabled}
            color={color}
            fontSize={fontSize}
            fontWeight={fontWeight}
            bgColor={bgColor}
            bgColorHover={bgColorHover}
            height={height}
            width={width}
            padding={padding}
            margin={margin}
            transition={transition}
            focusColor={focusColor}
            colors={colors}
            theme={theme}
            position={position}
            right={right}
        >
            {t(content)}
        </StButton>
    );
};

export default Button;
