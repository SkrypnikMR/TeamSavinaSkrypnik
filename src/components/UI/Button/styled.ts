import styled from 'styled-components';
import { IStyledButton } from './types';
import {
  bgColorDefault,
  colorDefault,
  focusColorDefault,
  disabledColor,
  bgColorDisabledDefault,
  disabledBorder,
  bgColorDefaultFon,
  bgMessage,
} from '../baseLayout';

export const StButton = styled.button < IStyledButton > `
  ${({ margin = '0 0 7px 0' }) => margin && `margin: ${margin}`};
  font-family: 'Play', sans-serif;
  padding: ${({ padding = '5px' }) => padding};
  width: ${({ width = '170px' }) => width};
  height: ${({ height = '35px' }) => height};
  ${({ cursor = 'pointer' }) => cursor && `cursor: ${cursor}`};
  color: ${({ color = colorDefault }) => color};
  background-color: ${({ bgColor = bgColorDefault }) => bgColor};
  border-radius: ${({ borderRadius = '7px' }) => borderRadius}; 
  border: ${({ border = 'none' }) => border};
  outline: ${({ outline = 'none' }) => outline};
  ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
  font-size: ${({ fontSize = '16px' }) => fontSize};
  position: ${({ position = 'static' }) => position};
  right: ${( right = '0px' ) => right};
  &:hover {
    background: ${({ bgColor = bgMessage }) => bgColor};
    color: ${({ colors, theme }) => colors[theme].buttonText};
    border: ${({ border = 'none' }) => border};
    }
  &:focus {
    color: ${({ focusColor = focusColorDefault }) => focusColor};
    }
  &:disabled{ 
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = bgColorDisabledDefault }) => bgColorDisabled};
    color: ${({ focusColor = disabledColor }) => focusColor};
    }
`;
