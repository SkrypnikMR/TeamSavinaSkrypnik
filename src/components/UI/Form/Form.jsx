import React from 'react';
import PropTypes from 'prop-types';
import { StFormDiv } from './styled';

const Form = ({ children }) => <StFormDiv>{children}</StFormDiv>;

Form.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]).isRequired,
};

export default Form;
