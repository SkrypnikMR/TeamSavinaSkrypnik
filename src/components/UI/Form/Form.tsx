import React from 'react';
import { StFormDiv } from './styled';
import { IProps } from './types';

const Form = ({ children } : IProps) => <StFormDiv>{children}</StFormDiv>;

export default Form;
