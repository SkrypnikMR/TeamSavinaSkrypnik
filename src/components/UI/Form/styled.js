import styled from 'styled-components';
import { colorDefault, TRANSPARENT_BACKGROUND } from '../baseLayout';

export const StFormDiv = styled.div`
    ${({ backgroundColor = TRANSPARENT_BACKGROUND }) => backgroundColor && `background-color: ${backgroundColor}`};
    min-height: 80vh;
    display: flex;
    width: 40%;
    margin: 20px auto 0 auto;
    justify-content: space-between;
    align-items: center;
    flex-direction: column;
    font-family: 'Play', sans-serif;
    border-radius: 20px;
    ${({ color = colorDefault }) => color && `color: ${color}`};
    ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
        p{
          font-size: 65px;
          padding: 0;
          margin: 10px 0 10px 0;
        }
        span{
          padding: 10px 10px 20px 10px;
        }
        a{
          padding: 0 10px;
          text-decoration: none;
          color: rgb(0, 0, 118);
          font-weight: bold;
          &:hover{
            text-decoration: underline;
          }
        }
        button{margin-bottom: 10px}
        @media (max-width: 920px){
            width: 100%;
            border-radius: 0px;
        }
`;
