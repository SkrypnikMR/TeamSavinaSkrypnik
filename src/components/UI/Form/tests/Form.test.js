import React from 'react';
import { shallow } from 'enzyme';
import Form from '../Form.jsx';
import Button from '../../Button';

describe('Form', () => {
    let props;
    beforeEach(() => {
        props = {
            children: [<Button id="1" onClick={jest.fn()} key={1} />],
        };
    });
    it('Should match snapshot', () => {
        const component = shallow(<Form {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StFormDiv', () => {
        const component = shallow(<Form {...props} />);
        expect(component.find('styled__StFormDiv')).toHaveLength(1);
    });
});
