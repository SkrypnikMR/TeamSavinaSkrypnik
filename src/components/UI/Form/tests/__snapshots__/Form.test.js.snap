// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match snapshot 1`] = `"<div class=\\"styled__StFormDiv-sc-1xa63f-0 hNLlQX\\"><button id=\\"1\\" type=\\"button\\" class=\\"styled__StButton-sc-1a6hck8-0 jZiUSN\\"></button></div>"`;
